internal/torcx/profile.go

```diff
@@ -157,15 +165,21 @@ func AddToProfile(profilePath string, im Image) error {
 		return err
 	}
 
+	// Try v1 first
+	if v1Profile, err := getProfileV1(profilePath); err == nil {
+		return addToProfileV1(profilePath, st.Mode().Perm(), v1Profile, &im)
+	}
+
+	// Fallback to v0
 	if v0Profile, err := getProfileV0(profilePath); err == nil {
 		return addToProfileV0(profilePath, st.Mode().Perm(), v0Profile, &im)
 	}
 
 	return errors.Wrapf(err, "unable to unmarshal profile to %s", profilePath)
```

Comment:
`err` is always nil here because this us using the `Stat` call's err, which has already been checked. The most readable solution might be to split the `if` statements into two lines instead of doing in-line assignments for them, thus making `err`'s scope last to this return.

Reply:
That was indeed poorly coded, I'm vastly grateful you caught it here! As suggested, I split the two oneliners in order to broaden last `err` scope.